### PR TITLE
Ignore comment lines in file_length rule

### LINKED ISSUE #1165: FileLengthRule rule should not count comment lines
If FunctionBodyLengthRule (#258) and TypeBodyLengthRule (#369) do it then so should FileLengthRule

### PR DESCRIPTION
**Pull request by sammy-SC**
Fixes #1165

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/FileLengthRule.swift [hunk 1]
```diff
@@ -18,15 +18,20 @@ public struct FileLengthRule: ConfigurationProviderRule, SourceKitFreeRule {
...
         triggeringExamples: [
-            repeatElement("//\n", count: 401).joined()
+            repeatElement("print(\"swiftlint\")\n", count: 401).joined()
         ]
     )
 
     public func validate(file: File) -> [StyleViolation] {
-        let lineCount = file.lines.count
+        let commentKinds = Set(SyntaxKind.commentKinds())
+        let lineCount = file.syntaxKindsByLines.filter { kinds in
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/FileLengthRule.swift]: probably we should only perform this filtering if the file would trigger a violation.

This reminds me of one of my first PRs and @jpsim's comment: https://github.com/realm/SwiftLint/pull/330#issuecomment-170160772

Comment (sammy-SC) [inline Source/SwiftLintFramework/Rules/FileLengthRule.swift]: @marcelofabri I changed it so it only performs filtering if the file would trigger a violation.

Let me know if you request any further changes.

**Comment by marcelofabri** [comment]
Sorry for taking so long to review this, but what do you think about making this a configuration (which would be disabled by default)?

**Comment by sammy-SC** [comment]
@marcelofabri I would turn this on by default for following reasons

- both `TypeBodyLengthRule` and `FunctionBodyLengthRule` don't have it configurable
- it will actually make rule less strict, which means that if it gets merged in this form, SwiftLint will not throw more warnings and errors, rather less.

If you insist on making it configurable I will be happy to implement it.

**Comment by vendruscolo** [comment]
Any updates on this? This is a feature I'd love to have 😄

**Comment by sammy-SC** [comment]
@marcelofabri I resolved the merge conflict, let me know if there is anything else you need me to do.

**Comment by jpsim** [comment]
This needs to be configurable, with the previous behavior enabled by default.

In other words, I'd see this as:

```yml
line_length:
  ignore_comment_only_lines: true
```

**Comment by sammy-SC** [comment]
@jpsim configuration added.

Let me know if there is anything you would like to me to change.

**Comment by jpsim** [comment]
Looks like the integration tests are failing: https://travis-ci.org/realm/SwiftLint/builds/252588434

```
Test Case '-[SwiftLintFrameworkTests.IntegrationTests testSwiftLintLints]' started.
fatal error: unexpectedly found nil while unwrapping an Optional value
Current stack trace:
0    libswiftCore.dylib                 0x00000001127da130 swift_reportError + 129
1    libswiftCore.dylib                 0x00000001127f6b50 _swift_stdlib_reportFatalError + 60
2    libswiftCore.dylib                 0x00000001125e7250 specialized specialized StaticString.withUTF8Buffer<A> ((UnsafeBufferPointer<UInt8>) -> A) -> A + 342
3    libswiftCore.dylib                 0x0000000112761e90 partial apply for (_fatalErrorMessage(StaticString, StaticString, file : StaticString, line : UInt, flags : UInt32) -> Never).(closure #2) + 109
4    libswiftCore.dylib                 0x00000001125e7250 specialized specialized StaticString.withUTF8Buffer<A> ((UnsafeBufferPointer<UInt8>) -> A) -> A + 342
5    libswiftCore.dylib                 0x00000001127149a0 specialized _fatalErrorMessage(StaticString, StaticString, file : StaticString, line : UInt, flags : UInt32) -> Never + 96
6    SwiftLintPackageTests              0x0000000111b13fd0 IntegrationTests.(testSwiftLintLints() -> ()).(closure #3).(closure #1) + 671
7    SwiftLintPackageTests              0x0000000111b167a0 partial apply for IntegrationTests.(testSwiftLintLints() -> ()).(closure #3).(closure #1) + 932
8    SwiftLintPackageTests              0x0000000111b15d80 String.(withStaticString((StaticString) -> ()) -> ()).(closure #1) + 151
9    SwiftLintPackageTests              0x0000000111b15f10 thunk + 35
10   SwiftLintPackageTests              0x0000000111b15fc0 partial apply for thunk + 89
11   libswiftCore.dylib                 0x000000011274ff80 partial apply for String.(withCString<A> ((UnsafePointer<Int8>) throws -> A) throws -> A).(closure #1) + 75
12   libswiftCore.dylib                 0x00000001125d85b0 _ContiguousArrayBuffer.withUnsafeBufferPointer<A> ((UnsafeBufferPointer<A>) throws -> A1) throws -> A1 + 111
13   libswiftCore.dylib                 0x00000001125ce380 ContiguousArray.withUnsafeBufferPointer<A> ((UnsafeBufferPointer<A>) throws -> A1) throws -> A1 + 88
14   libswiftCore.dylib                 0x0000000112626c20 String.withCString<A> ((UnsafePointer<Int8>) throws -> A) throws -> A + 142
15   SwiftLintPackageTests              0x0000000111b13e80 String.withStaticString((StaticString) -> ()) -> () + 287
16   SwiftLintPackageTests              0x0000000111b13900 IntegrationTests.(testSwiftLintLints() -> ()).(closure #3) + 1239
17   SwiftLintPackageTests              0x0000000111b14360 thunk + 589
18   libswiftCore.dylib                 0x00000001125b42e0 Sequence.forEach((A.Iterator.Element) throws -> ()) throws -> () + 404
19   SwiftLintPackageTests              0x0000000111b127f0 IntegrationTests.testSwiftLintLints() -> () + 1264
20   SwiftLintPackageTests              0x0000000111b146b0 @objc IntegrationTests.testSwiftLintLints() -> () + 34
21   CoreFoundation                     0x00007fffbce19040 __invoking___ + 140
22   CoreFoundation                     0x00007fffbce18e30 -[NSInvocation invoke] + 289
23   XCTest                             0x000000010e638cfa __24-[XCTestCase invokeTest]_block_invoke.234 + 50
24   XCTest                             0x000000010e67efc1 -[XCTMemoryChecker _assertInvalidObjectsDeallocatedAfterScope:] + 37
25   XCTest                             0x000000010e638657 __24-[XCTestCase invokeTest]_block_invoke_2 + 665
26   XCTest                             0x000000010e676b4c -[XCTestContext performInScope:] + 190
27   XCTest                             0x000000010e638546 -[XCTestCase invokeTest] + 254
28   XCTest                             0x000000010e638e7e -[XCTestCase performTest:] + 565
29   XCTest                             0x000000010e635efa __27-[XCTestSuite performTest:]_block_invoke + 300
30   XCTest                             0x000000010e635bf9 -[XCTestSuite _performProtectedSectionForTest:testSection:] + 29
31   XCTest                             0x000000010e635d26 -[XCTestSuite performTest:] + 214
32   XCTest                             0x000000010e635efa __27-[XCTestSuite performTest:]_block_invoke + 300
33   XCTest                             0x000000010e635bf9 -[XCTestSuite _performProtectedSectionForTest:testSection:] + 29
34   XCTest                             0x000000010e635d26 -[XCTestSuite performTest:] + 214
35   XCTest                             0x000000010e635efa __27-[XCTestSuite performTest:]_block_invoke + 300
36   XCTest                             0x000000010e635bf9 -[XCTestSuite _performProtectedSectionForTest:testSection:] + 29
37   XCTest                             0x000000010e635d26 -[XCTestSuite performTest:] + 214
38   XCTest                             0x000000010e689b24 __44-[XCTTestRunSession runTestsAndReturnError:]_block_invoke + 40
39   XCTest                             0x000000010e64c50c -[XCTestObservationCenter _observeTestExecutionForBlock:] + 587
40   XCTest                             0x000000010e6898d1 -[XCTTestRunSession runTestsAndReturnError:] + 281
41   XCTest                             0x000000010e6219ac -[XCTestDriver runTestsAndReturnError:] + 254
42   XCTest                             0x000000010e678fb8 _XCTestMain + 773
43   xctest                             0x000000010e61548d <unavailable> + 5261
44   libdyld.dylib                      0x000000010efb0234 start + 1
```

**Comment by sammy-SC** [comment]
@jpsim tests are fixed now.
I also ran `make sourcery` and simplified the code in `FileLenghtRule`.

**Comment by marcelofabri** [comment]
Continued on #1702.

**Pull request closed without merging**